internal/armada/scheduling/lease.go

```diff
@@ -325,6 +363,31 @@ func (c *leaseContext) leaseJobs(
 	return jobs, slice, nil
 }
 
+func isJobSchedulable(c *leaseContext, job *api.Job, remainder common.ComputeResourcesFloat, candidatesLimit LeasePayloadLimit) bool {
+	isJobLargeEnough := isLargeEnough(job, c.minimumJobSize)
+	isRemainderValid := remainder.IsValid()
+	isCandidateWithinLimit := candidatesLimit.IsWithinLimit(job)
+
+	isRegularlySchedulable := isJobLargeEnough && isRemainderValid && isCandidateWithinLimit
+	isPreemptiveJob := isJobLargeEnough && hasPriorityClass(job.PodSpec)
+
+	return isRegularlySchedulable || isPreemptiveJob
+}
+
+// validateOrDefaultPriorityClass checks is the pod spec's priority class configured as supported in Server config
+// if not, default to DefaultPriorityClass if it is specified
+// otherwise default to no Priority Class
+func validateOrDefaultPriorityClass(podSpec *v1.PodSpec, preemptionConfig configuration.PreemptionConfig) {
+	_, ok := preemptionConfig.PriorityClasses[podSpec.PriorityClassName]
+	if !ok {
+		if preemptionConfig.DefaultPriorityClass != "" {
+			podSpec.PriorityClassName = preemptionConfig.DefaultPriorityClass
+		} else {
+			podSpec.PriorityClassName = ""
+		}
```

Comment:
```suggestion
		podSpec.PriorityClassName = preemptionConfig.DefaultPriorityClass
```

Reply:
Both branches of the if statement produce the same result :)

Reply:
Clever is the arch enemy of clear, I like it :D